app/javascript/utils/form/hoc.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react'
```

Comment:
what is this filename? is this used at all in this PR? If it's going to be used, can you add some more documentation? e.g. even with an example of how to use it?

Reply:
Just adding this a reference https://reactjs.org/docs/higher-order-components.html
HOC are some helpers to wrap components to "Extend functionally" 
HOC usually creates a new component with the specified component wrapped